include/meta/meta_fwd.hpp

```diff
@@ -174,6 +174,24 @@
 #define META_IS_SAME(...) std::is_same<__VA_ARGS__>::value
 #endif
 
+#if defined(__GNUC__)
+#define META_IS_BASE_OF(...) __is_base_of(__VA_ARGS__)
```

Comment:
Clang and MSVC also support this intrinsic with the same name.

Reply:
Clang defines `__GNUC__`. I didn't know about MSVC, thanks.

Reply:
> Clang defines `__GNUC__`

Only in "pretending to be GCC" mode. In "pretending to be MSVC" mode it defines `_MSC_VER` instead. But yes, I think `#if defined(__GNUC__) || defined(__MSC_VER__)` covers all three.

Reply:
I didn't know that!